### DISCUSSION TITLE
osu! lazer's resolution is sometimes formatted to another connected display without consideration of the size/pixel density of said connected display

### DISCUSSION THREAD

**Discussion by megadrive7** [ppy/osu#21683] [Q&A]
### Type

Cosmetic

### Bug description

I play osu! with my DTC-133 tablet (which has a screen). I play with this screen on, along with my monitor, in duplicate screen mode. whenever the display is set to duplicate, osu!'s resolution is considerably worse. my best guess is that it is automatically formatted to the resolution of the tablet, which carries over to my main display. this low resolution is not fuzzy, but rather pixelated. this difference is either not present or unnoticeable minor on my tablet screen. turning my tablet display on after launching osu! lazer solves this problem. changing the resolution after launching with both displays on invariably results in a black screen crash. also, attempting to reopen osu! lazer at any point other than while initially launching it is impossible. the music will become focussed, the screen will go black, the music unfocussed, and it kicks me back to whatever window i was on. the only way to get back in is to close and reopen osu! lazer. this problem has been present ever since i started playing on osu!, but i never realized this bug until i launched osu! lazer with my tablet display off (i had assumed pixelated was a strange stylistic decision, and the opening bug a known problem that was a product of the game's state of production). Both displays are fullscreen.

### Screenshots or videos

![osu_2022-12-17_12-00-22](https://user-images.githubusercontent.com/112918046/208253004-53b29ced-ae69-4ac9-a54a-a34a0126bf15.jpg)

### Version

osu! 2022.1208.0-lazer

### Logs

[legacy-ipc.log](https://github.com/ppy/osu/files/10252211/legacy-ipc.log)
[network.log](https://github.com/ppy/osu/files/10252212/network.log)
[performance.log](https://github.com/ppy/osu/files/10252213/performance.log)
[runtime.log](https://github.com/ppy/osu/files/10252214/runtime.log)
[updater.log](https://github.com/ppy/osu/files/10252215/updater.log)
[database.log](https://github.com/ppy/osu/files/10252216/database.log)
[input.log](https://github.com/ppy/osu/files/10252217/input.log)

**Comment by Susko3** [comment]
> changing the resolution after launching with both displays on invariably results in a black screen crash.

Could you post the logs of this crash when setting resolution?

> also, attempting to reopen osu! lazer at any point other than while initially launching it is impossible. the music will become focussed, the screen will go black, the music unfocussed, and it kicks me back to whatever window i was on. the only way to get back in is to close and reopen osu! lazer.

Could you also post logs of this scenario? Close the game and open it, and reproduce this exactly.

Would also be helpful:
- Screenshot of the pixelated game taken with `Win`+`Print Screen` / just `Print Screen`
- Screenshot of Windows display settings with scaling and resolution.
- Try setting your monitors to 'extend' mode and see if you get the same (or different) problems.

**Reply by megadrive7** [reply]
i realized now that the photo i uploaded is higher in resolution than my actual display

and it seems we have uncovered another problem. using the screenshotting method suggested, all photos turn out like this.

![Screenshot (5)](https://user-images.githubusercontent.com/112918046/208254495-8069428e-eb64-4e72-83d8-3ff9ce9d7295.png)
this photo was taken while in song select

![Screenshot (7)](https://user-images.githubusercontent.com/112918046/208254507-d0b58602-b637-42d9-b26e-8b628015c9db.png)
this one was taken on the main menu

i should note that when i hover over the lazer icon in the tool bar while its running, i see the same thing pictured here

i have reproduced the second bug about being unable to reopen the game, and copied the logs with the game still open after. not sure if i shouldve done that with the game closed.

[runtime.log](https://github.com/ppy/osu/files/10252299/runtime.log)
[updater.log](https://github.com/ppy/osu/files/10252301/updater.log)
[database.log](https://github.com/ppy/osu/files/10252302/database.log)
[input.log](https://github.com/ppy/osu/files/10252303/input.log)
[legacy-ipc.log](https://github.com/ppy/osu/files/10252304/legacy-ipc.log)
[network.log](https://github.com/ppy/osu/files/10252305/network.log)
[performance.log](https://github.com/ppy/osu/files/10252306/performance.log)

i was unable to reproduce the resolution bug. for whatever reason, it occured every time i changed resolutions about an hour ago, but now does not occur. when i say blackscreen crash, i mean that osu! lazer was still open, selection noises for main menu items were still made, but both of my displays were blank. i would have to tab out and then close the game

extend seems to solve all of these problems, but i want to be able to use duplicate

**Reply by Susko3** [reply]
The game rendering at 1280x720, while the desktop resolution is 1920x1080 usually happens when your monitor(s) are using display scaling != 100%, in this case looks like it might be 150%. Can you confirm this?

Non-100% scaling breaking things are a common(ish) problem, but sometimes it works fine.
